pandas/io/stata.py

```diff
@@ -1580,13 +1580,13 @@ def _read_value_labels(self) -> None:
             # Don't read twice
             return
 
-        self._value_labels_read = True
         self._value_label_dict: dict[str, dict[int, str]] = {}
```

Comment:
Perhaps one last one. Should this be moved to the `__init__`? I prefer to declare all attributes there since it avoids late addition of attributes.

Reply:
I have now moved this up as suggested. I put it in the `# State variables for the file` section as this seemed the closest fit, but can shift it around it you like.